src/check_line.c

```diff
@@ -67,6 +44,32 @@ bool	ft_isvalid_float_block(char **line, float min, float max)
 	return (true);
 }
 
+bool	ft_isvalid_unit_vec(char **line, float min, float max)
+{
+	double	checksum;
+	size_t	tmp;
+	char	*begin;
+
+	begin = *line;
+	tmp = 0;
+	checksum = ft_strtod(*line, &tmp);
+	if (!ft_isvalid_float(line, min, max, true))
+		return (false);
+	(*line)++;
+	checksum += ft_strtod(*line, &tmp);
+	if (!ft_isvalid_float(line, min, max, true))
+		return (false);
+	(*line)++;
+	checksum += ft_strtod(*line, &tmp);
+	if (!ft_isvalid_float(line, min, max, false))
+		return (false);
+	if (checksum == 0.0)
+		return (ft_perror_not_unit(begin));
+	else if (checksum > 1.0)
```

Comment:
A unit vector can have a checksum > 1.0. The definition of a vectors length is

sqr(x²+y²+z²)

The unit vector will have 1 = x² +  y ² + z² (as 1 to the power of 2 is still 1). So the checksum of the individual components to the power of 2 should be checked against > 1.0

Example: 
(0.905538, 0.3, 0.3) is perfectly a unit vector, but its simple checksum is obviously > 1

Reply:
True, sloppy from me!
Adapted: catch the return and add it squared to the checksum.